dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -105,8 +105,19 @@ def show
 
     @script_level = ScriptLevelsController.get_script_level(@script, params)
     raise ActiveRecord::RecordNotFound unless @script_level
-    # If we have a signed out user for any of these cases we will want to redirect them to sign in
-    authenticate_user! if !can?(:read, @script) || @script.login_required? || (!params.nil? && params[:login_required] == "true")
+    if @script.login_required? || (!params.nil? && params[:login_required] == "true")
+      if cachable_request?(request)
+        # if login_required on a cached level, redirect to cached_page_auth_redirect
+        # See https://codedotorg.atlassian.net/browse/TEACH-758 for more details.
+        path_without_params = request.fullpath.split('?').first || request.fullpath
+        return redirect_to "/api/v1/users/cached_page_auth_redirect?user_return_to=" + path_without_params
```

Comment:
This seems like it will strip out all the original URL parameters. Understood that we don't want to forward the `login_required` param to avoid an infinite redirect; are there other params sometimes passed into cacheable levels that we might want to _not_ strip out?

Reply:
Yes, see my comment above, but that is something I would want to add. I want to get consensus on direction before I spend too much time on parsing strings to only remove the one parameter. My first attempt hit a couple snags due to permitting hash variables